enterprise/backend/src/metabase_enterprise/data_editing/api.clj

```diff
@@ -337,6 +337,24 @@
   (let [include-cols [:token :table_id :creator_id]]
     {:tokens (t2/select (into [:table_webhook_token] include-cols) :table_id table-id)}))
 
+(api.macros/defendpoint :get "/action/:action-id"
+  "Like /api/action but specialises the parameter requirements for a dashcard context where selected rows
+  maybe provided values for e.g primary key values.
+  Parameter names that intersect with the tables fields will be marked as isRowParam true.
+  Later when evaluated as a row action, you will be able to omit these parameters."
+  [{:keys [action-id]}   :- [:map [:action-id   :string]]
+   {:keys [dashcard-id]} :- [:map [:dashcard-id ms/PositiveInt]]]
+  (let [action             (-> (actions/select-action :id (parse-long action-id) :archived false)
+                               (t2/hydrate :creator)
+                               api/read-check)
+        {:keys [card_id]}  (api/check-404 (t2/select-one :model/DashboardCard dashcard-id))
+        {:keys [table_id]} (api/check-404 (t2/select-one :model/Card card_id))
+        fields             (t2/select [:model/Field :name] :table_id table_id)
+        field-names        (set (map :name fields))
+        specialize-param   (fn [param]
+                             (assoc param :isRowParam (contains? field-names (:slug param))))]
```

Comment:
How about rather just indicating which fields are PKs, and the rest of the row params just get marked as `:optional`?

Reply:
Hmm... Existing action parameters can already have `:required` key that is derived from the field `:database_required` for implicit actions, for SQL actions it is configured like this: 

![image](https://github.com/user-attachments/assets/1f32fc5b-770c-4815-931a-0f4be323fe88)

Reply:
Would you propose changing the value of `:required`?

Reply:
Dropping isRowParam as discussed, changed endpoint to `row-action. Now removing intersecting parameters from the list